### PR TITLE
Add sync committee client test coverage

### PR DESCRIPTION
**Pull request by julienharbulot**
T-ssv-037 — No Tests for Beacon Sync Committee APIs

Original ticket summary:
- `beacon/goclient/sync_committee.go:16-52` and `beacon/goclient/sync_committee_contribution.go:22-130` had no focused test coverage for `SyncCommitteeDuties`, `SubmitSyncMessages`, `GetSyncCommitteeContribution`, and `IsSyncCommitteeAggregator`.
- The ticket called out risk around parallel contribution fetching and the `modulo=0` aggregator edge case.

Changes:
- Add focused tests for sync committee duties, sync message submission, contribution fetching, and the zero-modulo aggregator path.
- Fix `GetSyncCommitteeContribution` to preallocate the result slice and write contributions by index, avoiding concurrent slice appends and preserving selection-proof alignment.

Fixes - Fixes ssvlabs/ssv-node-board#926

### CODE REVIEW

**Review by nkryuchkov — DISMISSED** [review]

File: beacon/goclient/sync_committee_test.go [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+			mu.Lock()
+			requested = append(requested, opts.SubcommitteeIndex)
+			mu.Unlock()
+
+			switch opts.SubcommitteeIndex {
+			case subnetIDs[0]:
+				time.Sleep(25 * time.Millisecond)
+			case subnetIDs[1]:
+				time.Sleep(5 * time.Millisecond)
+			}
```
[the remark this answers is not in the document]
Comment (julienh-ssv) [reply beacon/goclient/sync_committee_test.go]: Addressed in 3bc791fa9f34ba7fcecd074dced567b297cc4efd by replacing the time.Sleep-based ordering with a deterministic rendezvous using channels, so the test still exercises concurrent contribution fetches without wall-clock dependency.

[the remark this answers is not in the document]
Comment (julienh-ssv) [reply beacon/goclient/sync_committee_test.go]: Addressed in 3bc791fa9f34ba7fcecd074dced567b297cc4efd by replacing the time.Sleep-based ordering with a deterministic rendezvous using channels, so the test still exercises concurrent contribution fetches without wall-clock dependency.

**Review by nkryuchkov — APPROVED** [review]

File: beacon/goclient/sync_committee_test.go [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+			SyncCommitteeSize:                    1,
+			SyncCommitteeSubnetCount:             4,
+			TargetAggregatorsPerSyncSubcommittee: 1,
+		},
+	}
+
+	require.True(t, goClient.IsSyncCommitteeAggregator(make([]byte, 96)))
+}
+
+func signatureWithFirstByte(first byte) phase0.BLSSignature {
```
Comment (nkryuchkov) [inline beacon/goclient/sync_committee_test.go:317]: nit: do we need this? we could do `phase0.BLSSignature{first}`

**Review by iurii-ssv — DISMISSED** [review]

File: beacon/goclient/sync_committee_test.go (hunk 2 above)
Comment (julienh-ssv) [reply beacon/goclient/sync_committee_test.go:317]: Addressed in b968a4bdc6bdc8879934e87657d6f0caad91f9d7 by simplifying the helper to return the literal form directly: `phase0.BLSSignature{first}`.

**Review by iurii-ssv — APPROVED** [review]

**Review by ljuba-ssv — APPROVED** [review]

**Comment by Roy-blox** [comment]
## QA Verification — Sync-committee contribution ordering fix

Verified on stage-hoodi nodes 1-17, epochs 91788–91792 (2026-04-29).

**Zero reconstruction errors** across all observed sync-committee contribution duties. Submissions firing continuously with no errors on the SYNC_COMMITTEE_CONTRIBUTION runner. e2m miss rate (73-85%) within expected stage-hoodi range for the current sync committee period (512 → 672 assigned validators after period rollover).

Multi-subnet concurrent path (the core race condition) is covered by the PR's unit tests. Stage verification confirms single-subnet path healthy with no regressions.

**Result: PASS ✓**

Tracked in [QA-ssv-034 (#1035)](https://github.com/ssvlabs/ssv-node-board/issues/1035)
